src/main/settings.ts

```diff
@@ -287,7 +292,40 @@ export class Settings<T extends object> extends PropertyChangeEmitter<T> {
    */
   private saveSettings(updatedSettings: T) {
     this.watcher?.unwatch(this.filePath);
-    fs.writeJSONSync(this.filePath, updatedSettings, { spaces: 2 });
+    try {
+      fs.writeJSONSync(this.filePath, updatedSettings, { spaces: 2 });
+    } catch (error) {
+      // Handle read-only config files correctly.
+      if (error.code === 'EROFS' || error.code === 'EACCES') {
+        // Generate a temporary directory to write the files to for easy reference and write to it.
+        const tmpBaseDir = os.tmpdir() + '/kando/';
+        fs.mkdirSync(tmpBaseDir, { recursive: true });
+        const baseName = path.basename(this.filePath);
+        const tmpDir = tmpBaseDir + baseName;
+        fs.writeJSONSync(tmpDir, updatedSettings, { spaces: 2 });
+
+        // If the config option ignoreWriteProtectedConfigFiles is not set; notify the user that their hard work has not been permanently saved.
+        if (!this.ignoreWriteProtectedConfigFiles) {
+          const errorMessage =
+            'The ' +
+            baseName +
+            ' file was read-only. It will temporarily be saved to: ' +
+            tmpDir +
+            " Set ignoreWriteProtectedConfigFiles to 'true' to silence this warning";
+          console.log(errorMessage);
```

Comment:
I think you could use a warning here:

```suggestion
          console.warn(errorMessage);
```